### PR TITLE
Add support for Intel GPU to Word Language Model Example

### PR DESCRIPTION
**Pull request by framoncg**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,4 +22,5 @@ docs/venv
 
 # development
 .vscode
+.venv
```
Comment (dvrogozh) [inline .gitignore:25]: This might be specific to your personal setup. I think better to avoid doing this.

File: word_language_model/README.md [hunk 2]
```diff
@@ -5,10 +5,14 @@ The trained model can then be used by the generate script to generate new text.
 
 ```bash
 python main.py --cuda --epochs 6           # Train a LSTM on Wikitext-2 with CUDA.
+python main.py --xpu --epochs 6            # Train a LSTM on Wikitext-2 with Intel XPU.
```
Comment (dvrogozh) [inline word_language_model/README.md:8]: Drop. These cmdlines actually don't give any additional info. It's clear that you can use something else instead of `--cuda`.

File: word_language_model/README.md [hunk 3]
```diff
@@ -53,4 +58,5 @@ python main.py --cuda --emsize 650 --nhid 650 --dropout 0.5 --epochs 40
 python main.py --cuda --emsize 650 --nhid 650 --dropout 0.5 --epochs 40 --tied
 python main.py --cuda --emsize 1500 --nhid 1500 --dropout 0.65 --epochs 40
 python main.py --cuda --emsize 1500 --nhid 1500 --dropout 0.65 --epochs 40 --tied
+python main.py --xpu --emsize 1500 --nhid 1500 --dropout 0.65 --epochs 40 --tied
```
Comment (dvrogozh) [inline word_language_model/README.md:61]: Same - drop.

File: word_language_model/generate.py [hunk 4]
```diff
@@ -39,25 +56,38 @@
...
     parser.error("--temperature has to be greater or equal 1e-3.")
 
-with open(args.checkpoint, 'rb') as f:
-    model = torch.load(f, map_location=device)
-model.eval()
-
 corpus = data.Corpus(args.data)
 ntokens = len(corpus.dictionary)
 
+if args.model == 'Transformer':
```
Comment (dvrogozh) [inline word_language_model/generate.py:80]: This makes PR to go beyond declared scope - add XPU support. Can you separate this out to dedicated PR?

**Review by dvrogozh — CHANGES_REQUESTED** [review]

File: word_language_model/generate.py (hunk 4 above)
Comment (framoncg) [reply word_language_model/generate.py:80]: Should I leave this PR open and extract the changes (ie removing them from this one) that are separate from the XPU addition to open the other one before addressing the changes here?

Comment (dvrogozh) [reply word_language_model/generate.py:80]: As we discussed offline you can do a separate PR first for CUDA to fix existing example, wait for it merge and then make another PR to add XPU support on top. Alternatively you can just describe the changes you've made in the PR description paying attention why you are making them. I personally would go with the 1st option, but 2nd option is fine as well. You choice.

Comment (framoncg) [reply word_language_model/generate.py:80]: As per the suggestion I submitted the first PR regarding this set of changes here #10

**Comment by jafraustro** [comment]
Hi @framoncg, Can you close this PR as the discussion will continue in #10, Additionally, we need to update the example with the `accelerator` API
